admin/services/data-access/group.data.service.js

```diff
@@ -304,16 +304,39 @@ groupDataService.sqlMarkGroupAsDeleted = async (groupId) => {
 /**
  * Find groups by ids and school id.
  * @param schoolId
- * @param groupIds
+ * @param pupilIds
  * @returns {Promise<*>}
  */
-groupDataService.sqlFindGroupsByIds = async (schoolId, groupIds) => {
-  if (!schoolId || !groupIds || groupIds.length < 3) return false
-  const sql = `SELECT id, name 
+groupDataService.sqlFindGroupsByIds = async (schoolId, pupilIds) => {
+  if (!schoolId || !pupilIds || pupilIds.length < 3) return false
+
+  let sql = `SELECT id, name 
     FROM ${sqlService.adminSchema}.[group] 
-    WHERE school_id=${schoolId} 
-    AND id IN (${groupIds})`
-  const params = []
+    WHERE school_id=@schoolId`
+
+  let whereClause = ' AND id IN ('
```

Comment:
Please replace this where clause build out with a call to `sqlService.buildParameterList` which will reduce the amount of code in this function.